appcues/src/main/java/com/appcues/statemachine/StateMachine.kt

```diff
@@ -149,8 +155,10 @@ internal class StateMachine(
     private fun takeValidStateTransitions(state: State, action: Action) = with(Transitions) {
         when {
             // Idling
-            state is Idling && action is StartExperience ->
+            state is Idling && action is StartExperience -> {
+                startLifecycleTracking()
```

Comment:
I noticed that we were never really stopping lifecycle tracking before. So revised this - now we start tracking here when we notice any move from `Idling` to `StartExperience`. Then we stop tracking on L136 on any return to `Idling`.